lua/precognition/horizontal_motions.lua

```diff
@@ -0,0 +1,127 @@
+local utils = require("precognition.utils")
+
+local M = {}
+
+---@param str string
+---@return integer
+function M.line_start_non_whitespace(str)
+    return str:find("%S") or 0
+end
+
+---@param len integer
+---@return integer
+function M.line_end(len)
```

Comment:
Imo it'd make more sense to have a unified interface for motions. Maybe they can all take `(str, start, len)`?

Same goes for vertical motions but with a different interface.

Reply:
agree